contracts/truefi2/LoanFactory2.sol

```diff
@@ -178,8 +179,9 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
         return newToken;
     }
 
-    function isCreatedByFactory(IDebtToken loan) external override view returns (bool) {
-        return isLoanToken[loan] || isDebtToken[loan];
+    function isCreatedByFactory(IDebtToken debt) external override view returns (bool) {
```

Comment:
This is fine for now since we're about to split this function in a follow-up PR, but for the future I think we should consider typecasting and untyped addresses to be a code smell that indicates we're fighting the compiler. I'd rather duplicate code than leave it untyped to handle two different possible input types.